lib/metrics.coffee

```diff
@@ -79,6 +82,47 @@ module.exports =
     catch e
       createUUID()
 
+  # TODO: Remove these deprecation tracking methods after we remove 1.0 deprecations
+  watchDeprecations: ->
+    grim.on 'updated', (deprecation) =>
+      setImmediate =>
+        packageNames = {}
+        for stack in deprecation.getStacks()
+          packageName = stack.metadata?.packageName ? (@getPackageName(stack) or '').toLowerCase()
+          if packageName
+            pack = atom.packages.getLoadedPackage(packageName)
+            version = pack?.metadata?.version
+            packageNames[packageName] = version or 'unknown'
+
+        for packageName, version of packageNames
+          Reporter.sendEvent('deprecation', packageName, version) if packageName?
+
+        return
+
+  getPackageName: (stack) ->
+    resourcePath = atom.getLoadSettings().resourcePath
+    packagePaths = @getPackagePathsByPackageName()
+    for packageName, packagePath of packagePaths
+      if packagePath.indexOf('.atom/dev/packages') > -1 or packagePath.indexOf('.atom/packages') > -1
+        packagePaths[packageName] = fs.absolute(packagePath)
+
+    for i in [1...stack.length]
+      {functionName, location, fileName} = stack[i]
+      # Empty when it was run from the dev console
+      return unless fileName
+      for packageName, packagePath of packagePaths
+        relativePath = path.relative(packagePath, fileName)
+        return packageName unless /^\.\./.test(relativePath)
+      return "Your local #{path.basename(fileName)} file" if atom.getUserInitScriptPath() is fileName
```

Comment:
Do we want to report `~/.atom/init.coffee` deprecations?

Not sure if they are worth it since we can't actually fix them via a PR like we can for a package.